extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/JsonRegistryProcessor.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.micrometer.deployment.export;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.micrometer.deployment.MicrometerRegistryProviderBuildItem;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+import io.quarkus.micrometer.runtime.export.JsonMeterRegistryProvider;
+import io.quarkus.micrometer.runtime.export.JsonRecorder;
+import io.quarkus.micrometer.runtime.registry.json.JsonMeterRegistry;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+
+public class JsonRegistryProcessor {
+
+    private static final Logger log = Logger.getLogger(JsonRegistryProcessor.class);
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
```

Comment:
Execution of this step should still be guarded as others are.

Reply:
Why exactly is it useful to wrap it into a BooleanSupplier that checks whether the configuration enables it, instead of simply doing the same by an `if` condition in the method?